attentive-android-sdk/src/test/java/com/attentive/androidsdk/AttentiveApiTest.java

```diff
@@ -347,6 +349,38 @@ public void sendEvent_productViewEventWithAllParams_callsOkHttpClientWithCorrect
         assertEquals(addToCartItem.getProductVariantId(), m.getSubProductId());
     }
 
+    @Test
+    public void sendEvent_customEventWithAllParams_callsOkHttpClientWithCorrectPayload() throws JsonProcessingException {
+        // Arrange
+        givenAttentiveApiGetsGeoAdjustedDomainSuccessfully();
+        givenOkHttpClientReturnsSuccessFromEventsEndpoint();
+        CustomEvent customEvent = buildCustomEventWithAllFields();
+
+        // Act
+        attentiveApi.sendEvent(customEvent, ALL_USER_IDENTIFIERS, DOMAIN);
+
+        // Assert
+        ArgumentCaptor<Request> requestArgumentCaptor = ArgumentCaptor.forClass(Request.class);
+        verify(okHttpClient, times(1)).newCall(requestArgumentCaptor.capture());
+        Optional<Request> customEventRequest = requestArgumentCaptor.getAllValues().stream().filter(request -> request.url().toString().contains("t=ce")).findFirst();
+        assertTrue(customEventRequest.isPresent());
+        assertRequestMethodIsPost(customEventRequest.get());
+        HttpUrl url = customEventRequest.get().url();
+
+        assertEquals("modern", url.queryParameter("tag"));
+        assertEquals("mobile-app", url.queryParameter("v"));
+        assertEquals("0", url.queryParameter("lt"));
+        assertEquals(GEO_ADJUSTED_DOMAIN, url.queryParameter("c"));
+        assertEquals("ce", url.queryParameter("t"));
+        assertEquals(ALL_USER_IDENTIFIERS.getVisitorId(), url.queryParameter("u"));
```

Comment:
broke out some of this stuff into helper functions in the automated testing branch - might be worth merging main into this branch & using that helper method?

Reply:
can we also add this event to the AttentiveApiTestIT?

Reply:
Good call. The helper function was in AttentiveApiTestIT - I added a similar helper function to AttentiveApiTest and refactored all relevant the tests to use the new helper function.

Also added the event to AttentiveApiTestIT 👍